src/trading/ClaimProceeds.sol

```diff
@@ -0,0 +1,84 @@
+pragma solidity ^0.4.13;
+
+import 'ROOT/Controller.sol';
+import 'ROOT/Mutex.sol';
+import 'ROOT/reporting/Interfaces.sol';
+import 'ROOT/extensions/MarketFeeCalculator.sol';
+import 'ROOT/legacy_reputation/SafeMath.sol';
+
+
+// AUDIT: Ensure that a malicious market can't subversively cause share tokens to be paid out incorrectly.
+/**
+ * @title ClaimProceeds
+ * @dev This allows users to claim their money from a market by exchanging their shares
+ */
+contract ClaimProceeds is Controlled {
+    using SafeMath for uint256;
+
+    function publicClaimProceeds(IMarket _market) onlyInGoodTimes public returns(bool) {
+        Mutex _mutex = Mutex(controller.lookup("Mutex"));
+        _mutex.acquire();
+
+        require(_market.isFinalized());
+        require(block.timestamp > _market.getFinalizationTime() + 3 days);
+
+        ReportingToken _winningReportingToken = _market.getFinalWinningReportingToken();
+
+        for (uint8 _outcome = 0; _outcome < _market.getNumberOfOutcomes(); ++_outcome) {
+            IShareToken _shareToken = _market.getShareToken(_outcome);
+            uint256 _numberOfShares = _shareToken.balanceOf(msg.sender);
+            var (_shareHolderShare, _creatorShare, _reporterShare) = divideUpWinnings(_market, _winningReportingToken, _outcome, _numberOfShares);
+
+            // always destroy shares as it gives a minor gas refund and is good for the network
+            if (_numberOfShares > 0) {
+                _shareToken.destroyShares(msg.sender, _numberOfShares);
+            }
+            ERC20 _denominationToken = _market.getDenominationToken();
+            // NOTE: rounding error here will result in _very_ tiny amounts of denominationToken left in the market
```

Comment:
This comment should be moved onto `calculateProceeds` since that is where the problem actually lies.